mmlearn/tasks/classification.py

```diff
@@ -0,0 +1,250 @@
+        self.metrics: Dict[Any, Metric] = {}
+
+        self.tokenizer = tokenizer
+
+    def on_evaluation_epoch_start(
+        self, pl_module: LightningModule, all_dataset_info: Dict[str, Any]
+    ) -> None:
+        """Move the metrics to the device of the Lightning module."""
+        for metric in self.metrics.values():
+            metric.to(pl_module.device)
+
+        if self.mode == "zeroshot":
+
+            class LabelDescriptionDataset(Dataset):
+                def __init__(
+                    self, descriptions: List[str], tokenizer: PreTrainedTokenizerBase
+                ) -> None:
+                    self.descriptions = descriptions
+                    self.tokenizer = tokenizer
+
+                def __len__(self) -> int:
+                    return len(self.descriptions)
+
+                def __getitem__(self, idx: int) -> Example:
+                    description = self.descriptions[idx]
+                    tokens = self.tokenizer(description)
+
+                    example = Example(
+                        {
+                            Modalities.RGB: torch.rand(3, 224, 224),
+                            Modalities.TEXT: description,
+                        }
+                    )
+
+                    if tokens is not None:
+                        if isinstance(tokens, dict):  # output of HFTokenizer
+                            assert (
+                                Modalities.TEXT in tokens
+                            ), f"Missing key `{Modalities.TEXT}` in tokens."
+                            example.update(tokens)
+                        else:
+                            example[Modalities.TEXT] = tokens
+
+                    return example
+
+            if self.tokenizer is None:
+                raise ValueError(
+                    "Tokenizer must be set in the dataset to generate tokenized label descriptions"
+                )
+
+            # Set the label embeddings
+            for name, dataset_info in all_dataset_info.items():
+                descriptions = [
+                    "This image has a sign of " + label
+                    for label in dataset_info.get_label_mapping().values()
+                ]
+
+                dataset = LabelDescriptionDataset(descriptions, self.tokenizer)
+                batch_size = len(dataset)
+                dataloader = DataLoader(
+                    dataset,
+                    batch_size=batch_size,
+                    shuffle=False,
+                    collate_fn=collate_example_list,
+                )
+                batch = next(iter(dataloader))
+                batch = {
+                    key: value.to(pl_module.device) if torch.is_tensor(value) else value
+                    for key, value in batch.items()
+                }
+                all_dataset_info[name].set_label_embedding(
+                    pl_module(batch)[Modalities.get_modality(Modalities.TEXT).embedding]
+                )
```

Comment:
I don't think all of this is necessary. You could take the class labels as input, tokenize them and get their embedding without having to define a dataset class or dataloader.

Reply:
Done

Reply:
The only thing is that to give the descriptions to the model, I also need to pass `Modalities.RGB` to the model as well (if I have not missed anything)

How about if I define a dummy image in constants?